task_3/main.cpp

```diff
@@ -0,0 +1,86 @@
+#include "counter.h"
+
+#include <iostream>
+#include <boost/bind.hpp>
+#include <boost/asio.hpp>
+#include <boost/thread/thread.hpp>
+#include <experimental/filesystem>
+#include <chrono>
+#include <queue>
+#include <fstream>
+#include <string>
+
+using std::string;
+using std::cout;
+using std::endl;
+using namespace std::experimental::filesystem;
+
+extern boost::mutex global_stream_lock;
+void WorkerThread( boost::shared_ptr< boost::asio::io_service > io_service );
+
+int main( int argc, char * argv[] ) {
+	path dir {argc > 1 ? argv[1] : current_path()};				//	getting initial path
+        if (!exists(dir)) {
+        	cout << "Path " << dir << " does not exist.\n";
+        	return 1;
+	}
+	string outFile {argc > 2 ? argv[2] : "statistics.txt"};		//	getting output file name and # of threads
```

Comment:
The same is applied to the filename. User doesn't know if he have to type extension or so. Also if user will launch the program without arguments he could probably loose important data, previously stored in the statistics.txt

Reply:
agree with it